jobmon_client/src/jobmon/client/task_template.py

```diff
@@ -600,7 +610,7 @@ def create_task(
             op_args=op_args,
             compute_resources=compute_resources,
             compute_resources_callable=compute_resources_callable,
-            resource_scales=resource_scales,
+            resource_scales=copy.deepcopy(resource_scales),
```

Comment:
could be worthwhile to check for valid types here. i.e. users can provide an iterator here but not lists according to your implementation, seems like an easy mistake to make that will lead to confusing errors later

The docstring you added is very nice, but let's be honest people probably won't read it :)

Reply:
Will do!